### PR TITLE
JAMES-3807 Split webadmin-http-client in a separate maven module than webadmin-cli

### PR DESCRIPTION
**Pull request by vttranlina**
https://www.mail-archive.com/server-dev@james.apache.org/msg72269.html

### CODE REVIEW

File: server/pom.xml [hunk 1]
```diff
@@ -49,6 +49,7 @@
         <module>apps/scaling-pulsar-smtp</module>
         <module>apps/spring-app</module>
         <module>apps/webadmin-cli</module>
+        <module>apps/webadmin-http-client</module>
```
Comment (chibenwa) [inline server/pom.xml:52]: I would prefer to have it in `server/protocols/webadmin/webadmin-http-client`. After all this is not an application (no Main)

**Review by Arsnael — APPROVED** [review]
Looks good to me

**Comment by chibenwa** [comment]
Can we please open a JIRA ticket for this?

(I can add it to the commits when merging).

**Comment by quantranhong1999** [comment]
> Can we please open a JIRA ticket for this?

https://issues.apache.org/jira/browse/JAMES-3807

File: pom.xml [hunk 2]
```diff
@@ -2772,6 +2772,11 @@
                 <artifactId>mockito-core</artifactId>
                 <version>4.6.0</version>
             </dependency>
+            <dependency>
+                <groupId>org.mockito</groupId>
+                <artifactId>mockito-inline</artifactId>
+                <version>4.7.0</version>
+            </dependency>
```
Comment (quantranhong1999) [inline pom.xml:2779]: Should we parameterize both `mockito-core` and `mockito-inline` versions (should be the same also)?

**Review by quantranhong1999 — APPROVED** [review]

**Pull request merged**